yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -426,6 +432,33 @@ export class AztecNodeService implements AztecNode {
     return await Promise.all(leafValues.map(leafValue => committedDb.findLeafIndex(treeId, leafValue.toBuffer())));
   }
 
+  public async findLeavesIndexesWithApproxBlockNumber(
```

Comment:
I profoundly hate everything about this, but we need to attach a blockNumber to a nullifier, so we're able to track when a note was destroyed and can resurrect them if a reorg happens.

Here `minBlockNumber` refers to "how far back I'm willing to go to attach a blockNumber to this nullifier if it exists in the tree, because I don't care if it's so old it's never gonna be reorge'd"

Before adding clarifying comments, improving it to be a binary search, or any more work of the kind, any chance this can be replaced soon-ish by a dedicated method in the world state? @spalladino (also CC @alexghr since we discussed this and he provided the algorithm)

Reply:
I don't think this belongs in the world state. By all means a separate store in the node/archiver that maps nullifiers to the block they were published in. 

It's effectively a convenience cache for clients much like logs.

Reply:
@PhilWindle not 100% a fan of that approach, but I will do it like that. My rationale is that we might want/need to access this info in the future for other trees, not only the nullifier tree AND world state already contains a notion of `blockNumber` (or rather "revision"), so it doesn't feel like a stretch for it to provide the metadata.

Reply:
As discussed, I mis-understood what you guys need for this. We can add a method to query the block number for a given index on a given tree quite easily.